rust/src/format/page_table.rs

```diff
@@ -68,7 +82,47 @@ impl PageTable {
         Ok(Self { pages })
     }
 
-    pub fn set_page_info(&mut self) {}
+    pub async fn write(&self, writer: &mut ObjectWriter) -> Result<usize> {
+        let pos = writer.tell();
+        assert!(!self.pages.is_empty());
+        let num_columns = self.pages.keys().max().unwrap() + 1;
+        let num_batches = self
```

Comment:
page_table.num_columns and page_table.num_batches?

Reply:
not sure wdym?

PageTable does not have these two fields.